kernel/aster-nix/src/process/process/mod.rs

```diff
@@ -98,31 +110,57 @@ impl Process {
         sig_dispositions: Arc<Mutex<SigDispositions>>,
         resource_limits: ResourceLimits,
         nice: Nice,
-    ) -> Self {
+    ) -> Arc<Self> {
         let children_pauser = {
             // SIGCHID does not interrupt pauser. Child process will
             // resume paused parent when doing exit.
             let sigmask = SigMask::from(SIGCHLD);
             Pauser::new_with_mask(sigmask)
         };
 
-        Self {
-            pid,
-            threads: Mutex::new(threads),
-            executable_path: RwLock::new(executable_path),
-            process_vm,
-            children_pauser,
-            status: Mutex::new(ProcessStatus::Uninit),
-            parent: Mutex::new(parent),
-            children: Mutex::new(BTreeMap::new()),
-            process_group: Mutex::new(Weak::new()),
-            file_table,
-            fs,
-            umask,
-            sig_dispositions,
-            resource_limits: Mutex::new(resource_limits),
-            nice: Atomic::new(nice),
-        }
+        let process = Arc::new_cyclic(|process_ref: &Weak<Process>| {
+            let current_process = process_ref.clone();
+            let sent_signal = move || {
+                let signal = KernelSignal::new(SIGALRM);
+                if let Some(process) = current_process.upgrade() {
+                    process.enqueue_signal(signal);
+                }
+            };
```

Comment:
Currently I group the logic to a single block. I think it may not be suitable to group all the logic to a single function for the reason that the signal-send-related closure will be reused for the creation of `virtual timer` and `prof timer` in `Process`, and the `Process` needs to store `prof clock` which will be used to create `prof timer`. At this time, the returned value (three timer and a clock) of this function will be weired. So I think it may be better to use a clock to group the logic or use a function with callback closure and based clock as parameters, and the created timer as returned value (but the content of such function will be very simple).